src/fsharp/pars.fsy

```diff
@@ -2440,7 +2440,8 @@ attrUnionCaseDecl:
   | opt_attributes opt_access unionCaseName EQUALS constant
       { if Option.isSome $2 then errorR(Error(FSComp.SR.parsEnumFieldsCannotHaveVisibilityDeclarations(), rhs parseState 2))
         let mDecl = rhs2 parseState 1 5
-        (fun xmlDoc -> Choice1Of2 (SynEnumCase ( $1, $3, $5, xmlDoc, mDecl))) } 
+        let mValue = rhs parseState 5
```

Comment:
I had a thought: instead of having to calculate the range of the value N times for each case here, what if you changed the return type of the `constant` rule so that it returns a tuple `SynConst * Range`?  That way the range is calculated as close to the point of 'capture' as possible, and can be used/discarded at the callsites as necessary.

Reply:
That works as well, good catch.
Any idea when to use `lhs` or `rhs`? Or doesn't that matter all that much?